### PR TITLE
improved code efficiency

### PR DESCRIPTION
**Pull request by jchpan**

### CODE REVIEW

File: appyters/dimensionality_reduction_visualization/dimensionality_reduction_visualization.ipynb [hunk 1]
```diff
@@ -372,11 +353,14 @@
...
     "# Normalize Data\n",
+    "def log2_normalize(x, offset=1.):\n",
+    "    return np.log2(x + offset)\n",
+    "\n",
     "df_data_norm = log2_normalize(expr_df, offset=1)\n",
     "df_data_norm = qnorm.quantile_normalize(df_data_norm, axis=0)\n",
     "\n",
     "#convert to zscores\n",
-    "expr_df = pd.DataFrame(zscore(df_data_norm, axis=1), index=df_data_norm.index, columns=df_data_norm.columns)\n",
+    "expr_df = pd.DataFrame(zscore(df_data_norm, axis=0), index=df_data_norm.index, columns=df_data_norm.columns)\n",
```
Comment (u8sand) [inline appyters/dimensionality_reduction_visualization/dimensionality_reduction_visualization.ipynb:363]: I'm a little confused about the orientation. is the dataframe genes on rows/samples on columns or the other way around

File: appyters/dimensionality_reduction_visualization/dimensionality_reduction_visualization.ipynb [hunk 2]
```diff
@@ -469,24 +431,31 @@
    "source": [
     "%%appyter code_exec\n",
     "# Data transformation for 2D visualization\n",
-    "\n",
     "{% if visualization_method.value == \"All\" or visualization_method.value == \"PCA\" %}\n",
-    "pca = PCA(n_components=min(clustering_topk, len(expr_df)))\n",
-    "transformed_pca = pca.fit_transform(expr_df)\n",
-    "pca_df = create_df(transformed_pca, df_y)\n",
+    "# PCA results have been computed when doing Leiden \n",
+    "pca_df = df_y.copy()\n",
```
Comment (u8sand) [inline appyters/dimensionality_reduction_visualization/dimensionality_reduction_visualization.ipynb:436]: Can we try and eliminate these copies? Can't you just either use `leiden_df.obsm['']` directly or reasign the variable, i.e. `pca_df = leiden_df.obsm['X_pca']`

**Review by u8sand — CHANGES_REQUESTED** [review]

File: appyters/dimensionality_reduction_visualization/dimensionality_reduction_visualization.ipynb (hunk 1 above)
Comment (jchpan) [reply appyters/dimensionality_reduction_visualization/dimensionality_reduction_visualization.ipynb:363]: The dataframe has columns to be gene. In this case, do we set axis to be 0?
Screen Shot 2022-12-01 at 1 14 23 PM

Comment (u8sand) [reply appyters/dimensionality_reduction_visualization/dimensionality_reduction_visualization.ipynb:363]: The axis is correct if this is the orientation. Though we should use the other orientation for display since the standard is the other way.

Comment (jchpan) [reply appyters/dimensionality_reduction_visualization/dimensionality_reduction_visualization.ipynb:363]: Sure. I will display the data input before the transpose.

**Review by u8sand — APPROVED** [review]
